### PR TITLE
Move locale files into new folder structure

### PR DESCRIPTION
**Pull request by Nitemaeric**
## Context

Some locales were added into the old structure after the locale structure was updated.

## Changes proposed in this pull request

- Move locale files around.
- Add relative namespaced locale lookup for FormObjects and ViewComponents. `forms` and `components`, respectively.
- Update locales for error pages.

## Guidance to review

- Moved locale keys nesting matches directory nesting.
- The `support_email` keys feel like configuration values, not locales, but we want to hold off from adding in https://github.com/rubyconfig/config for now.

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated or [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by JamieCleare2525** [comment]
-
```
en.yml
en:
  shared:
    search_form:
      search-header: Search by organisation name or postcode
      search: Search
      clear_search: Clear search
```
this should be moved to `en/shared/search_form.yml`

-
```
en.yml
en:
  ... (lines 111 - 152)
  claims:
    support:
      users:
        index:
          ...
```
this should be moved to `en/claims/support/users.yml`

- Might be worth moving the `en: components:` from `en.yml` to a `en/components.yml`

**Review by JamieCleare2525 — COMMENTED** [review]
-
```
en.yml
en:
  shared:
    search_form:
      search-header: Search by organisation name or postcode
      search: Search
      clear_search: Clear search
```
this should be moved to `en/shared/search_form.yml`

-
```
en.yml
en:
  ... (lines 111 - 152)
  claims:
    support:
      users:
        index:
          ...
```
this should be moved to `en/claims/support/users.yml`

- Might be worth moving the `en: components:` from `en.yml` to a `en/components.yml`

**Comment by Nitemaeric** [comment]
@JamieCleare2525 👍 I'll pick this up again Monday morning :)

File: app/views/errors/internal_server_error.html.erb [hunk 1]
```diff
@@ -1,18 +1,19 @@
 <%= content_for :page_title, "Sorry, there’s a problem with the service" %>
```
Comment (JamieCleare2525) [inline app/views/errors/internal_server_error.html.erb:1]: Should the text in this file be in a locale?

File: app/views/errors/not_found.html.erb [hunk 2]
```diff
@@ -1,19 +1,19 @@
 <%= content_for :page_title, "Page not found" %>
```
Comment (JamieCleare2525) [inline app/views/errors/not_found.html.erb:1]: Should the text in this file be in a locale?

File: app/views/errors/unprocessable_entity.html.erb [hunk 3]
```diff
@@ -1,14 +1,15 @@
 <%= content_for :page_title, "Sorry, there’s a problem with the service" %>
```
Comment (JamieCleare2525) [inline app/views/errors/unprocessable_entity.html.erb:1]: Should the text in this file be in a locale?

File: app/views/errors/internal_server_error.html.erb (hunk 1 above)
Comment (Nitemaeric) [reply app/views/errors/internal_server_error.html.erb]: Addressed in https://github.com/DFE-Digital/itt-mentor-services/pull/139/commits/91ee412b43ad855c42892dc57bd5f475bf809fa5

File: app/views/errors/not_found.html.erb (hunk 2 above)
Comment (Nitemaeric) [reply app/views/errors/not_found.html.erb]: Addressed in https://github.com/DFE-Digital/itt-mentor-services/pull/139/commits/91ee412b43ad855c42892dc57bd5f475bf809fa5

File: app/views/errors/unprocessable_entity.html.erb (hunk 3 above)
Comment (Nitemaeric) [reply app/views/errors/unprocessable_entity.html.erb]: Addressed in https://github.com/DFE-Digital/itt-mentor-services/pull/139/commits/91ee412b43ad855c42892dc57bd5f475bf809fa5

**Review by JamieCleare2525 — APPROVED** [review]
LGTM! Nicely Formatted

File: app/components/application_component.rb [hunk 4]
```diff
@@ -1,5 +1,7 @@
 class ApplicationComponent < GovukComponent::Base
   def initialize(classes: [], html_attributes: {})
+    @virtual_path = "components/#{self.class.name.underscore}"
```
Comment (CatalinVoineag) [inline app/components/application_component.rb:3]: Nice

**Review by CatalinVoineag — APPROVED** [review]
Looks good

**Pull request merged**